### PR TITLE
Make Expo work with Gatsby SSR

### PR DESCRIPTION
**Pull request by slorber**
The current Gatsby + Expo demo didn't work so well with SSR (gatsby production build).
This PR fixes it and allow to SSR the expo examples

I had to do some workarounds but at least I have something published to show:
https://gatsby-expo-test.netlify.com

See also https://github.com/slorber/gatsby-plugin-react-native-web/issues/23

### CODE REVIEW

**Comment by slorber** [comment]
Thanks for the updates I needed on SSR @EvanBacon , will update the PR when I can (holiday soon).

What do you think about creating a package with universal expo examples? 
So that we could run / publish the same examples more easily on:
- expo --web
- next
- gatsby
- others...

File: with-gatsby/babel.config.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 module.exports = function(api) {
   api.cache(true);
   return {
-    presets: ['babel-preset-expo'],
+    presets: ["babel-preset-expo"],
+    plugins: [["babel-plugin-remove-graphql-queries",{},"test-name"]]
```
Comment (slorber) [inline with-gatsby/babel.config.js]: @EvanBacon , in Gatsby world it's not very common to have a babelrc or babel config at the root of the project.

But for Gatsby it seems required to use the "babel-plugin-remove-graphql-queries" (at least unless I figure out how to merge the Gatsby/Expo js loaders into a single one...)

It would be convenient if you exposed me a way to customize the default babel loader here and add the required gatsby plugin without creating a local babel config.

Basically, I want my Gatsby plugin to add the babel plugin here: https://github.com/expo/expo-cli/blob/f48370005f016f4b8700b31da9f1d51b4ac142a0/packages/webpack-config/src/loaders/createBabelLoader.ts#L139

Comment (EvanBacon) [reply with-gatsby/babel.config.js]: Not the cleanest API, but it is possible to customize the babel loader by finding it and modifying it like so https://github.com/software-mansion/react-native-gesture-handler/blob/master/webpack.config.js

Comment (slorber) [reply with-gatsby/babel.config.js]: yeah will try that. There is likely to be 2 jsloaders and I should just modify one of them.
That's a bit fragile but will probably work

Comment (slorber) [reply with-gatsby/babel.config.js]: that seems to work: https://github.com/slorber/gatsby-plugin-react-native-web/commit/663754ae2748ece5d6be11f19558bc2451a8f4a6

**Comment by slorber** [comment]
Hi,

Just made an update:
- used latest rngh to fix hammerjs bug SSR, and put back the RNGH example
- updated the "useREM" hook to fix the SSR bug
- automatically added the required gatsby plugin to expo js loader so that user does not need to

Also updated the netlify deployment with new build if you want to check: https://gatsby-expo-test.netlify.com/expo


That starts to become usable.

Do you want to change something else before merging?

**Comment by slorber** [comment]
One last annoying thing is that people may want to use the gatsby plugin, and benefit from RN libs transpilation setup of expo webpack config, without using Expo web apis.

For example, if someone is using just RNW + RN paper, without Expo, the webpack config works fine for that. But unfortunately that makes a warning: https://github.com/slorber/gatsby-plugin-react-native-web/issues/21

Do you think you could remove that warning for web?

**Comment by EvanBacon** [comment]
🚧 Stale, closing. Feel free to reopen when updated.
